classes/script_metadata.php

```diff
@@ -224,6 +224,44 @@ public static function get_request(): string {
         return $request;
     }
 
+    /**
+     * Gets the name of the script.
+     * Parameters are share the names of globals as a hint to the caller.
+     * @param ?string $ME
+     * @param ?string $SCRIPT
+     * @return string the request path for this profile.
+     */
+    public static function get_normalised_relative_script_path(?string $ME, ?string $SCRIPT): string {
```

Comment:
this makes a new method but doens't remove the old one get_request which should now be unused code after this commit?

Can you please also squash this commit with the ones which come after it as they are all logically part of the same change and then we should be good to go